caluma-demo/app/routes/application.js

```diff
@@ -1,12 +1,38 @@
 import { inject as service } from "@ember/service";
 import Route from "@ember/routing/route";
+import config from "../config/environment";
 
+const { languages, fallbackLanguage } = config;
 export default class ApplicationRoute extends Route {
   @service intl;
 
-  beforeModel() {
-    super.beforeModel(...arguments);
+  languages;
+  fallbackLanguage;
 
-    return this.intl.setLocale("en");
+  getBrowserLanguage() {
+    return (navigator.languages || [navigator.language])
+      .map(locale => locale)
+      .find(lang => languages.includes(lang));
+  }
+
+  getLocalStorageLanguage() {
+    if (languages.includes(localStorage.getItem("language"))) {
+      return localStorage.getItem("language");
+    }
+  }
+
+  getLanguage() {
+    return (
+      this.getLocalStorageLanguage() ||
+      this.getBrowserLanguage() ||
+      fallbackLanguage
+    );
+  }
+
+  beforeModel(transition) {
+    this.intl.setLocale([
+      this.getLanguage(transition.to.queryParams.language),
```

Comment:
`queryParamLanguage` was removed from the `getLanguage` function, so this is no longer correct.